static/src/components/performancestats/confusion_matrix.vue

```diff
@@ -0,0 +1,134 @@
+<template>
+    <div>
+        <h4>Precision-Recall By Label</h4>
+        <div class="form-inline">
+            <div class="form-group">
+                <label for="beta">Beta: </label>
+                <input id="beta" type="text" class="form-control input-sm" v-model="beta">
+            </div>
+        </div>
+        <table class="table table-bordered">
+            <thead>
+                <tr>
+                    <th class="col-xs-4">Label</th>
+                    <th class="col-xs-2">Precision</th>
+                    <th class="col-xs-2">Recall</th>
+                    <th class="col-xs-2">F1</th>
+                </tr>
+            </thead>
+            <tbody>
+                <tr v-for="(label, ix) in labels" :key="label">
+                    <th>{{label}}</th>
+                    <td>{{fmtNum(precision[ix])}}</td>
+                    <td>{{fmtNum(recall[ix])}}</td>
+                    <td>{{fmtNum(f1[ix])}}</td>
+                </tr>
+                <tr>
+                    <th>Average</th>
+                    <td v-for="(val, i) in averages" :key="i">{{fmtNum(val)}}</td>
+                </tr>
+            </tbody>
+        </table>
+        <h4>Confusion Matrix</h4>
+        <div id="matrix"></div>
+    </div>
+</template>
+<script>
+import Matrix from './matrix';
+
+function vectorSum(v, w) {
+    return v.map((x, ix) => x + w[ix]);
+}
+
+function matrixSum(a, b) {
+    return a.map((v, ix) => vectorSum(v, b[ix]));
+}
+
+export default {
+    props: ['stats', 'labels'],
```

Comment:
One thing that we can start setting the props like: a: {
      type: Number,
      required: true
    },   I'm planning to include in the lint rules this plugin plugin:vue/strongly-recommended

I think it helps to understand what the components are doing plus you can also set default values in case you want to.

What do you think?

Reply:
Yes, that's good - I'll change this